### PR TITLE
Add DenseNet on architectures

### PR DESCRIPTION
**Pull request by HenryMantilla**
### Description

Add fully functional DenseNet model implementation with parameter count verification.

### Checklist
Ensure your pull request meets the following requirements:
- [x] I have reviewed existing notebooks to avoid duplications.
- [x] My notebook is fully executed.
- [x] I have included a list of all libraries and their versions used in the notebook.
- [x] The notebook file is named clearly to reflect the topic _(e.g., `cross-entropy-loss.ipynb`)._
- [x] I have placed the notebook in the appropriate directory under `notebooks/`.
- [x] My code is simple, precise, and adheres to good coding practices.

### CODE REVIEW

File: README.md
Comment (guillepinto) [inline README.md:1]: please do not modify the README.md, instead add the information about your notebook in `automation/notebooks-table-data.csv` following the [CONTRIBUTING.md](https://github.com/semilleroCV/deep-learning-notes/blob/main/CONTRIBUTING.md#4-update-the-csv) instructions and the bot will take care of modifying the README.md

**Review by guillepinto — CHANGES_REQUESTED** [review]
good notebook!

File: README.md
Comment (HenryMantilla) [reply README.md:1]: New commit made, direct modification of README.md removed.

**Review by guillepinto — APPROVED** [review]
Good!

**Pull request merged**